### PR TITLE
vmi-create-admitter_test refactor: refactor probe context tests

### PR DESCRIPTION
**Pull request by Acedus**
### What this PR does
**Before this PR:**

The tests in the "with probes given" context used instantiated VMIs using functions that didn't leverage `libvmi` as well as didn't make use of the `newAdmissionReviewForVMICreation` function to create admission reviews.

**After this PR:**

The tests in the "with probes given" context now instantiate VMIs using libvmi and use the `newAdmissionReviewForVMICreation` function to create admission reviews.

Partially addresses: https://github.com/kubevirt/kubevirt/issues/13407

### Checklist

This checklist is not enforcing, but it's a reminder of items that could be relevant to every PR.
Approvers are expected to review this list.

- [ ] Design: A [design document](https://github.com/kubevirt/community/tree/main/design-proposals) was considered and is present (link) or not required
- [ ] PR: The PR description is expressive enough and will help future contributors
- [ ] Code: [Write code that humans can understand](https://en.wikiquote.org/wiki/Martin_Fowler#code-for-humans) and [Keep it simple](https://en.wikipedia.org/wiki/KISS_principle)
- [ ] Refactor: You have [left the code cleaner than you found it (Boy Scout Rule)](https://learning.oreilly.com/library/view/97-things-every/9780596809515/ch08.html)
- [ ] Upgrade: Impact of this change on upgrade flows was considered and addressed if required
- [ ] Testing: New code requires [new unit tests](https://github.com/kubevirt/kubevirt/blob/main/docs/reviewer-guide.md#when-is-a-pr-good-enough). New features and bug fixes require at least on e2e test
- [ ] Documentation: A [user-guide update](https://github.com/kubevirt/user-guide/) was considered and is present (link) or not required. You want a user-guide update if it's a user facing feature / API change.
- [ ] Community: Announcement to [kubevirt-dev](https://groups.google.com/g/kubevirt-dev/) was considered

### Release note

```release-note
NONE
```

### CODE REVIEW

File: pkg/virt-api/webhooks/validating-webhook/admitters/vmi-create-admitter_test.go [hunk 1]
```diff
@@ -286,19 +289,20 @@ var _ = Describe("Validating VMICreate Admitter", func() {
...
-					TCPSocket: &k8sv1.TCPSocketAction{Host: "lal", Port: intstr.Parse("80")},
-				},
-			}
-			vmi.Spec.LivenessProbe = &v1.Probe{
-				InitialDelaySeconds: 2,
-				Handler: v1.Handler{
-					HTTPGet: &k8sv1.HTTPGetAction{Host: "test", Port: intstr.Parse("80")},
-				},
-			}
+			vmi := newBaseVmi(
```
Comment (orelmisan) [inline pkg/virt-api/webhooks/validating-webhook/admitters/vmi-create-admitter_test.go:292]: nit: Please consider adding `libvmi.WithAutoAttachPodInterface(false)`, as it is required in order to reach this scenario in practice.

**Review by orelmisan — APPROVED** [review]
Thank you @Acedus

File: pkg/virt-api/webhooks/validating-webhook/admitters/vmi-create-admitter_test.go (hunk 1 above)
Comment (Acedus) [reply pkg/virt-api/webhooks/validating-webhook/admitters/vmi-create-admitter_test.go:292]: Good catch, thanks!

**Comment by Acedus** [comment]
[Change](https://github.com/kubevirt/kubevirt/compare/88769ebe8264d97a9b8664e3a8cbaaa3f9836d99..e8ce160f643028be83833c3603a19028ddf6acda) - Address @orelmisan comments.

**Review by orelmisan — APPROVED** [review]

**Review by jean-edouard — COMMENTED** [review]
/approve

**Pull request merged**